landlab/data_record/data_record.py

```diff
@@ -14,40 +15,70 @@ class DataRecord(object):
     Dataset by providing additional attributes and functions, including the
     ability to aggregate values on Landlab grid elements.
 
-    Data variables can vary along one or both of the following dimensions:
+    DataRecord uses the concept of an "item", a physical thing that is located
+    on the grid and has some properties, stored as variables. DataRecord was
+    built to keep track of variables associated with a Landlab grid across
+    items, through time, or both.
+
+    Thus data variables can vary along one or both of the following dimensions:
         - time (model time)
         - item_id: variables can characterize a set of items (each identified
             by an individual id) that reside on the grid.
 
+    If an item or set of items is defined, each item must be defined by the
+    grid element and the element id at which it resides, e.g.:
+
+        grid_element = 'node'
+        element_id = 9.
+
+    When items are defined, each item is given a unique id and the underlying
+    Dataset uses a dimension "item_id".
+
     Examples:
         - the variable 'mean_elevation' characterizes the grid and varies with
             time,
         - the variable 'clast__rock_type' characterizes a set of items (clasts)
             and varies with item_id,
         - the variable 'clast__size' can vary with both time and item_id
 
-    If an item or set of items is defined, each item must be defined by the
-    grid element and the element id at which it resides, e.g.:
-
-        grid_element = 'node'
-        element_id = 9.
-
     In the above case, `grid_element` and `element_id` are default data
     variables (in addition to any user-specified variables).
 
     For each item, `element_id` must be less than the number of this item's
-    grid_element that exist on the grid. For example, if the grid has 100
-    links, no item can live at link 100 or link -3 because only links 0 to 99
-    exist in this example.
+    grid_element that exist on the grid or be one of the dummy element values.
+    For example, if the grid has 100 links, and no dummy link values are
+    indicated, then, no item can live at link 100 or link -3 because only links
+    0 to 99 exist in this example.
+
+    Anything that the DataRecord keeps track of is considered a "record",
+    whether it uses one or both of the two standard dimensions.
+
+    DataRecord provides two method to assist with adding new records. The
+    method ``add_item`` should be used when no new variables are being added.
+    The method ``add_record`` should be used when new variables are being added.
```

Comment:
I **think** I've appropriately represented the difference between add_record and add_item. Let me know if you think it is incorrect.

Reply:
@margauxmouchene commented on this.